core/src/main/java/org/apache/iceberg/avro/ValueWriters.java

```diff
@@ -135,145 +150,145 @@ private NullWriter() {
 
     @Override
     public void write(Void ignored, Encoder encoder) throws IOException {
-      encoder.writeNull();
+      throw new IllegalStateException("[BUG] NullWriter shouldn't be used for writing nulls for Avro");
```

Comment:
Is this change necessary? Do you think that it would actually cause a problem if it were used?

Reply:
I think it won't, this was just me trying to make this fail loudly instead of fail silently and to avoid usage confusion in code, I can revert it back if you think it's not necessary.

Reply:
I think we should revert this back so that `writeNull` is correctly called. Nulls shouldn't be written as anything, but the fact that the encoder has a `writeNull` method makes me think that it should be called. We don't know what the encoder might be using it for.